### PR TITLE
docs: add some undocumented debug methods

### PR DESCRIPTION
**Pull request by s1na**
See https://github.com/ethereum/go-ethereum/issues/24808

### CODE REVIEW

File: docs/_rpc/ns-debug.md [hunk 1]
```diff
@@ -664,3 +664,102 @@ flag.
...
+Flattens the entire key-value database into a single level, removing all unused slots and merging all keys.
+
+| Client  | Method invocation                                  |
+|:--------|----------------------------------------------------|
+| Console | `debug.chaindbCompact()`                           |
+| RPC     | `{"method": "debug_chaindbCompact", "params": []}` |
+
+### debug_getHeaderRlp
+
+Retrieves the RLP encoded for of a single header.
```
Comment (holiman) [inline docs/_rpc/ns-debug.md:697]:
```suggestion
Returns an RLP-encoded header.
```

File: docs/_rpc/ns-debug.md [hunk 2]
```diff
@@ -664,3 +664,102 @@ flag.
...
+
+| Client  | Method invocation                                   |
+|:--------|-----------------------------------------------------|
+| Console | `debug.getHeaderRlp(blockNum)`                      |
+| RPC     | `{"method": "debug_getHeaderRlp", "params": [num]}` |
+
+### debug_accountRange
+
+Enumerates all accounts at a given block with paging capability. `maxResults` are returned in the page and the items have keys that come after the `start` key (hashed address).If `incompletes` is false, then accounts for which the key preimage doesn't exist in db are skipped.
```
Comment (holiman) [inline docs/_rpc/ns-debug.md:706]:
```suggestion
Enumerates all accounts at a given block with paging capability. `maxResults` are returned in the page and the items have keys that come after the `start` key (hashed address). 

If `incompletes` is false, then accounts for which the key preimage (i.e: the `address`) doesn't exist in db are skipped. NB: geth by default does not store preimages.  


```

**Comment by holiman** [comment]
I was thinking -- wouldn't it be nicer to place the docs in the code, on the corresponding methods, and then generate this text base off of that?

**Comment by s1na** [comment]
Yes that'd be much nicer. Currently this is happening for some of the methods in the go docs, e.g. https://pkg.go.dev/github.com/ethereum/go-ethereum@v1.10.17/eth#PublicDebugAPI.AccountRange. But for example all the methods under `internal` are skipped. There should be a way to use the same pipeline to generate the content and somehow add that to the website with a script.

**Comment by Danyalkasiri** [comment]
در تاریخ دوشنبه ۹ مهٔ ۲۰۲۲،‏ ۱۲:۴۸ Sina Mahmoodi ***@***.***>
نوشت:

**Comment by s1na** [comment]
To review you can look at this [commit](https://github.com/ethereum/go-ethereum/pull/24827/commits/9d5b95d38e12c65cd74777a94bc3db9bcbffc172). The second commit only changes ordering of methods to sort them alphabetically.

**Pull request merged**